File: ios/ExpoAdaptiveGlassSwitchModule.swift
import ExpoModulesCore

public class ExpoAdaptiveGlassSwitchModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoAdaptiveGlassSwitch")

    View(GlassSwitchView.self) {
      Events("onValueChange")

      Prop("value") { (view, value: Bool) in view.value = value }
      Prop("disabled") { (view, value: Bool) in view.disabled = value }
      Prop("onColor") { (view, value: UIColor?) in view.onColor = value }
      Prop("lens") { (view, value: Bool) in view.lens = value }
      // Android only, iOS follows the window's style
      Prop("scheme") { (_: GlassSwitchView, _: String) in }

      OnViewDidUpdateProps { view in
        view.propsDidUpdate()
      }
    }
  }
}

File: android/src/main/java/expo/modules/adaptiveglass/ExpoAdaptiveGlassSwitchModule.kt
package expo.modules.adaptiveglass

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoAdaptiveGlassSwitchModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoAdaptiveGlassSwitch")

    View(GlassSwitchView::class) {
      Events("onValueChange")

      Prop("value") { view: GlassSwitchView, value: Boolean -> view.value = value }
      Prop("disabled") { view: GlassSwitchView, value: Boolean -> view.disabled = value }
      Prop("onColor") { view: GlassSwitchView, value: Int? -> view.onColor = value }
      Prop("lens") { view: GlassSwitchView, value: Boolean -> view.lens = value }
      Prop("scheme") { view: GlassSwitchView, value: String -> view.scheme = value }

      OnViewDidUpdateProps { view: GlassSwitchView ->
        view.propsDidUpdate()
      }
    }
  }
}
